pkg/deployment/cert.go

```diff
@@ -116,8 +116,9 @@ func EnsureTLSCerts(ctx context.Context, helper *helper.Helper,
 			issuerLabelSelector = map[string]string{service.Spec.TLSCert.Issuer: ""}
 		}
 
-		issuer, err = certmanager.GetIssuerByLabels(ctx, helper, "", issuerLabelSelector)
+		issuer, err = certmanager.GetIssuerByLabels(ctx, helper, instance.Namespace, issuerLabelSelector)
 		if err != nil {
+			helper.GetLogger().Info("Error retrieving issuer by label", "issuerLabelSelector", issuerLabelSelector)
```

Comment:
+1 this will help with future debugging

ade and i briefly chatted about the fact this should be detected as part of the nodeset reconciliation not the deployment reconciliation

moving the validation fo the issuers is out of scope of this pr.